code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingInternal.java

```diff
@@ -208,22 +236,25 @@ public void error(final ExtensionError extensionError) {
 
             if (EventType.GENERIC_IDENTITY.getName().equalsIgnoreCase(eventToProcess.getType()) &&
                     EventSource.REQUEST_CONTENT.getName().equalsIgnoreCase(eventToProcess.getSource())) {
-
                 // handle the push token from generic identity request content event
                 handlePushToken(eventToProcess);
             } else if (MessagingConstant.EventType.MESSAGING.equalsIgnoreCase(eventToProcess.getType()) &&
                     EventSource.REQUEST_CONTENT.getName().equalsIgnoreCase(eventToProcess.getSource())) {
-
                 // Need experience event dataset id for sending the push token
                 if (!configSharedState.containsKey(MessagingConstant.SharedState.Configuration.EXPERIENCE_EVENT_DATASET_ID)) {
                     Log.warning(LOG_TAG, "%s - Unable to track push notification interaction, experience event dataset id is empty. Check the messaging launch extension to add the experience event dataset.", SELF_TAG);
                     return;
                 }
-
                 // handle the push tracking information from messaging request content event
                 handleTrackingInfo(eventToProcess);
+            // validate the edge response event from Offers then load any rules present
+            } else if (MessagingConstant.EventType.EDGE.equalsIgnoreCase(eventToProcess.getType()) &&
```

Comment:
why no `isEdgePersonalizationEvent` method here like we have for the others? 😄

Reply:
hmmm, i must have missed it. it will be added 😄.